doc/nrf/drivers/mspi_sqspi.rst

```diff
@@ -0,0 +1,192 @@
+on which the sQSPI is to operate. These two things are supplied in properties
+of the node labelled ``cpuflpr_vpr``.
+It is also important to delete the ``execution-memory`` property in the
+``cpuflpr_vpr`` node if it exists, so that the Nordic VPR coprocessor launcher
+is not used for the FLPR core (see :kconfig:option:`CONFIG_NORDIC_VPR_LAUNCHER`).
+
+Details on RAM configuration for the SoCs supported by sQSPI can be found in:
+- ::ref:`nrf54L15_porting_guide_ram_configuration`
+- ::ref:`nrf54H20_porting_guide_ram_configuration`
+
+The RAM region reserved for the sQSPI needs to be labelled ``softperiph_ram``
+on nRF54H20, so that it can be set as non-cacheable in the MPU configuration.
+
+See the following configuration example for the nRF54L15 SoC:
+
+.. code-block:: devicetree
+
+	&pinctrl {
+		sqspi_default: sqspi_default {
+			group1 {
+				psels = <NRF_PSEL(SDP_MSPI_SCK, 2, 1)>,
+					<NRF_PSEL(SDP_MSPI_CS0, 2, 5)>,
+					<NRF_PSEL(SDP_MSPI_DQ0, 2, 2)>;
+				nordic,drive-mode = <NRF_DRIVE_E0E1>;
+			};
+			group2 {
+				psels = <NRF_PSEL(SDP_MSPI_DQ1, 2, 4)>,
+					<NRF_PSEL(SDP_MSPI_DQ2, 2, 3)>,
+					<NRF_PSEL(SDP_MSPI_DQ3, 2, 0)>;
+				nordic,drive-mode = <NRF_DRIVE_E0E1>;
+				bias-pull-up;
+			};
+		};
+
+		sqspi_sleep: sqspi_sleep {
+			group1 {
+				low-power-enable;
+				psels = <NRF_PSEL(SDP_MSPI_SCK, 2, 1)>,
+					<NRF_PSEL(SDP_MSPI_CS0, 2, 5)>,
+					<NRF_PSEL(SDP_MSPI_DQ0, 2, 2)>,
+					<NRF_PSEL(SDP_MSPI_DQ1, 2, 4)>,
+					<NRF_PSEL(SDP_MSPI_DQ2, 2, 3)>,
+					<NRF_PSEL(SDP_MSPI_DQ3, 2, 0)>;
+			};
+		};
+	};
+
+	&cpuflpr_vpr {
+		pinctrl-0 = <&sqspi_default>;
+		pinctrl-1 = <&sqspi_sleep>;
+		pinctrl-names = "default", "sleep";
+		interrupts = <76 NRF_DEFAULT_IRQ_PRIORITY>;
+		status = "okay";
+	};
+
+	/ {
+		reserved-memory {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
+			softperiph_ram: memory@2003c000 {
+				reg = <0x2003c000 0x4000>;
+				ranges = <0 0x2003c000 0x4000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				sqspi: sqspi@3c00 {
+					compatible = "nordic,nrf-sqspi";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <0x3c00 0x200>;
+					status = "okay";
+					zephyr,pm-device-runtime-auto;
+				};
+			};
+		};
+	};
+
+For the nRF54H20 SoC, additionally a region for data buffers to be used in
+sQSPI transfers must be reserved in RAM accessible by the FLPR core. This
+region needs to be indicated to the shim driver by the ``memory-regions``
+property in the node representing the sQSPI. If an MSPI transfer will be
+requested with a buffer located outside of this region, the shim driver will
+allocate a buffer in the region for the time of the transfer and do the proper
+copying of data between these two buffers (before a TX transfer or after an RX
+transfer). To avoid the overhead of this automatic allocation and copying,
+a proper buffer can be allocated statically using :c:macro:`DMM_MEMORY_SECTION`.
+Then the shim driver will pass such buffer directly to the sQSPI.
```

Comment:
```suggestion
For the nRF54H20 SoC, you must also reserve a RAM region for data buffers used in
sQSPI transfers, accessible by the FLPR core. 
Indicate this region to the shim driver using the ``memory-regions`` property in the sQSPI node. 
If you initiate an MSPI transfer with a buffer outside this region, the shim driver will temporarily allocate a buffer within the region and correctly transfer data between the two buffers - before transmitting (TX) or after receiving (RX).
To avoid overhead from automatic allocation and copying, allocate buffers statically using :c:macro:`DMM_MEMORY_SECTION`. 
The shim driver will then directly pass these buffers to the sQSPI.
```